tosproc: completely remove compiler execution

alaviss opened the pull request:
Previously this test would depend on the compiler being available to execute to recompile itself.

Not only is this hugely inefficient, but it also produce a dependency on the compiler being in `PATH`.

Instead this pull request:

- Fold external runners into the same test binary. This removes the need of recompilation altogether.
- Replaced `nim r` in tests with much simpler "consume all" runner that allow testing the same thing. This prevents compiler bugs from tripping this test up.

Done as part of #89 to remove `PATH` dependency from test suite.

saem commented on tests/stdlib/tosproc.nim:
Should we try to add repo qualifiers for the bug references?

saem commented on tests/stdlib/tosproc.nim:
```diff
@@ -4,60 +4,60 @@ joinable: false
 #[
 joinable: false
-because it'd need cleanup up stdout
+because it has to execute itself
```
I think this would be better if the runner supported an array argument sets to pass a test so it could call it repeatedly instead. Although in this case we'd need more for the whole input output bit? Also scope creep.

alaviss replied:
This test tests `osproc`, and it executes itself with `osproc.*` to run its subprogram to verify behaviors. It has nothing to do with having to run with different arguments :P

saem replied:
Wait, the sub-programs are itself, right? Or did I read that wrong?

Anyhow, I was considering it for each of the permutation that it would run, would still need to have other processes and all that. Just that the reporting could be a little bit more fine grained.

alaviss replied:
Ah yea, we can certainly add a runner <-> test IPC in the future.

saem reviewed (approved):
Minor comments, I think I get it. It's better than before, that's for sure.

alaviss committed:
tosproc: completely remove compiler execution

Previously this test would depend on the compiler being available to
execute to recompile itself.

Not only is this hugely inefficient, but it also produce a dependency on
the compiler being in PATH.

Instead:

  * Fold external runners into the same test binary. This removes the
    need of recompilation altogether.

  * Replaced `nim r` in tests with much simpler "consume all" runner that
    allow testing the same thing. This prevents compiler bugs from
    tripping this test up.

alaviss commented:
bors r+
